ingester/src/deferred_load.rs

```diff
@@ -163,10 +164,24 @@ where
     /// The background task will wait a uniformly random duration of time
     /// between `[0, max_wait)` before attempting to pre-fetch `T` by executing
     /// the provided future.
-    pub(crate) fn new<F>(max_wait: Duration, loader: F) -> Self
+    pub(crate) fn new<F>(max_wait: Duration, loader: F, metrics: Arc<metric::Registry>) -> Self
```

Comment:
This is a minor, but because this type doesn't retain a reference to the registry after returning from this function, we can accept a reference to it, "borrowing" it from the caller for a while:

```suggestion
    pub(crate) fn new<F>(max_wait: Duration, loader: F, metrics: &metric::Registry) -> Self
```

An `Arc<T>` will dereference to a `&T`, so you can pass in the `Arc<metric::Registry>` as a `&metric::Registry` - you'll find most metric wrappers do it like this, because it is both semantically clearer ("I do not hold onto this") and also marginally more performant, because there's no need to +1 and -1 the ref counter via the `Arc::clone()` call, which gives the compiler's optimiser the best picture of what is intended.